src/renderer/components/modal/swap/SwapStateStepper.tsx

```diff
@@ -1,62 +1,48 @@
-import {
-  DbStatePathType,
-  isMergedBtcCancelledDbState,
-  isMergedBtcLockedDbState,
-  isMergedBtcRedeemedDbState,
-  isMergedCancelTimelockExpiredDbState,
-  isMergedDoneBtcPunishedDbState,
-  isMergedDoneBtcRefundedDbState,
-  isMergedDoneXmrRedeemedDbState,
-  isMergedEncSigSentDbState,
-  isMergedExecutionSetupDoneDbState,
-  isMergedXmrLockedDbState,
-  isMergedXmrLockProofReceivedDbState,
-  MergedDbState,
-  getTypeOfPathDbState,
-} from 'models/databaseModel';
-import { useActiveDbState, useAppSelector } from '../../../../store/hooks';
+import { DbStatePathType, getTypeOfPathDbState } from 'models/databaseModel';
+import { useActiveSwapInfo, useAppSelector } from '../../../../store/hooks';
+import { SwapStateName } from '../../../../models/rpcModel';
 
 function getActiveStep(
-  dbState: MergedDbState | null,
+  stateName: SwapStateName | null,
   processExited: boolean
 ): [DbStatePathType, number, boolean] {
-  if (dbState) {
-    const pathType = getTypeOfPathDbState(dbState);
+  if (stateName) {
+    const pathType = getTypeOfPathDbState(stateName);
 
     if (pathType === DbStatePathType.HAPPY_PATH) {
-      if (isMergedExecutionSetupDoneDbState(dbState)) {
+      if (stateName === SwapStateName.SwapSetupCompleted) {
         return [pathType, 0, processExited];
       }
-      if (isMergedBtcLockedDbState(dbState)) {
+      if (stateName === SwapStateName.BtcLocked) {
         return [pathType, 0, processExited];
       }
-      if (isMergedXmrLockProofReceivedDbState(dbState)) {
+      if (stateName === SwapStateName.XmrLockProofReceived) {
         return [pathType, 1, processExited];
       }
-      if (isMergedXmrLockedDbState(dbState)) {
+      if (stateName === SwapStateName.XmrLocked) {
         return [pathType, 2, processExited];
       }
-      if (isMergedEncSigSentDbState(dbState)) {
+      if (stateName === SwapStateName.EncSigSent) {
         return [pathType, 2, processExited];
       }
-      if (isMergedBtcRedeemedDbState(dbState)) {
+      if (stateName === SwapStateName.BtcRedeemed) {
         return [pathType, 3, processExited];
       }
-      if (isMergedDoneXmrRedeemedDbState(dbState)) {
+      if (stateName === SwapStateName.XmrRedeemed) {
         return [pathType, 4, false];
       }
     } else {
-      if (isMergedCancelTimelockExpiredDbState(dbState)) {
+      if (stateName === SwapStateName.CancelTimelockExpired) {
         return [pathType, 0, processExited];
       }
-      if (isMergedBtcCancelledDbState(dbState)) {
+      if (stateName === SwapStateName.BtcCancelled) {
         return [pathType, 1, processExited];
       }
-      if (isMergedDoneBtcRefundedDbState(dbState)) {
+      if (stateName === SwapStateName.BtcRefunded) {
         return [pathType, 2, false];
       }
-      if (isMergedDoneBtcPunishedDbState(dbState)) {
+      if (stateName === SwapStateName.BtcPunished) {
         return [pathType, 1, true];
       }
     }
```

Comment:
**Suggestion:** The `getActiveStep` function seems to have a lot of repeated code. Consider refactoring it to reduce redundancy and improve readability.
```suggestion
function getActiveStep(
  stateName: SwapStateName | null,
  processExited: boolean
): [DbStatePathType, number, boolean] {
  if (stateName) {
    const pathType = getTypeOfPathDbState(stateName);
    const stepMap = {
      [SwapStateName.SwapSetupCompleted]: 0,
      [SwapStateName.BtcLocked]: 0,
      // ... more mappings ...
    };
    return [pathType, stepMap[stateName] || 0, processExited];
  }
}
```